client/my-sites/theme/main.jsx

```diff
@@ -723,6 +727,31 @@ class ThemeSheet extends Component {
 		return <div>{ this.props.description }</div>;
 	};
 
+	renderNotice = () => {
+		const { activeThemeId, name, siteIntent, translate } = this.props;
+		const isAIAssembler = siteIntent === SiteIntent.AIAssembler && activeThemeId === 'assembler';
+		if ( ! isAIAssembler ) {
```

Comment:
Can we also check if the theme detail page is not the Assembler? Otherwise, there might be an edge case where the user activates the Assembler theme and sees this notice.